tests/crowdsourcing/tasks/turn_annotations_static/test_turn_annotations_static.py

```diff
@@ -0,0 +1,139 @@
+"""
+End-to-end testing for the chat demo crowdsourcing task.
+"""
+
+import json
+import os
+import unittest
+from typing import List
+
+
+TASK_CONFIG_FOLDER = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), 'task_config'
+)
+EXPECTED_STATES_FOLDER = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), 'expected_states'
+)
+
+
+try:
+
+    from parlai.crowdsourcing.tasks.turn_annotations_static.run import TASK_DIRECTORY
+    from parlai.crowdsourcing.tasks.turn_annotations_static.turn_annotations_blueprint import (
+        STATIC_BLUEPRINT_TYPE,
+        STATIC_IN_FLIGHT_QA_BLUEPRINT_TYPE,
+    )
+    from parlai.crowdsourcing.tasks.turn_annotations_static.util import build_task
+    from parlai.crowdsourcing.utils.tests import AbstractOneTurnCrowdsourcingTest
+
+    class TestTurnAnnotationsStatic(AbstractOneTurnCrowdsourcingTest):
+        """
+        Test the turn annotations crowdsourcing tasks.
+        """
+
+        def setUp(self):
+            super().setUp()
+            build_task(task_directory=TASK_DIRECTORY)
+
+        def test_no_in_flight_qa(self):
+            """
+            Test static turn annotations without in-flight QA.
+            """
+            overrides = [
+                '+mephisto.blueprint.annotation_indices_jsonl=null',
+                f'mephisto.blueprint.data_jsonl={TASK_CONFIG_FOLDER}/sample_conversations.jsonl',
+            ]
+            self._test_turn_annotations_static_task(
+                blueprint_type=STATIC_BLUEPRINT_TYPE,
+                expected_state_path=os.path.join(
+                    EXPECTED_STATES_FOLDER, 'no_in_flight_qa.json'
+                ),
+                overrides=overrides,
+            )
+
+        def test_in_flight_qa(self):
+            """
+            Test static turn annotations with in-flight QA.
+            """
+            overrides = [
+                '+mephisto.blueprint.annotation_indices_jsonl=null',
+                f'mephisto.blueprint.data_jsonl={TASK_CONFIG_FOLDER}/sample_conversations.jsonl',
+                f'+mephisto.blueprint.onboarding_in_flight_data={TASK_DIRECTORY}/task_config/onboarding_in_flight.jsonl',
+            ]
+            self._test_turn_annotations_static_task(
+                blueprint_type=STATIC_IN_FLIGHT_QA_BLUEPRINT_TYPE,
+                expected_state_path=os.path.join(
+                    EXPECTED_STATES_FOLDER, 'in_flight_qa.json'
+                ),
+                overrides=overrides,
+            )
+
+        def test_in_flight_qa_annotation_file(self):
+            """
+            Test static turn annotations with in-flight QA and with an annotation file.
+
+            The annotation file will list which turns of which conversations should
+            receive annotations.
+            """
+            overrides = [
+                f'+mephisto.blueprint.annotation_indices_jsonl={TASK_DIRECTORY}/task_config/annotation_indices_example.jsonl',
```

Comment:
nit: Is there a place for doing sanity check on  the indices jsonl  and can raise index out of range error before the actual mephisto job is alive or otherwise it would output error at the front end?

Reply:
This is something that should happen in `TurnAnnotationsStaticBlueprint.assert_task_args` (causing the system to fail on invalid input before Mephisto creates anything at all), however this method has not been implemented yet in `TurnAnnotationsStaticBlueprint`. It will still fail before a task is launched to workers in `__init__` though, as this method is called before launching tasks and it relies on annotation_indices_jsonl being valid.